org.osgi.test.common/src/main/java/org/osgi/test/common/service/ServiceConfiguration.java

```diff
@@ -125,36 +178,75 @@ public S getService(ServiceReference<S> reference) {
 
 	@Override
 	public long getTimeout() {
-		return trackServices.getTimeout();
+		return timeout;
 	}
 
 	@Override
 	public int getTrackingCount() {
-		return getTracker().getTrackingCount();
+		return tracker.getTrackingCount();
 	}
 
 	@Override
 	public SortedMap<ServiceReference<S>, S> getTracked() {
-		return getTracker().getTracked();
+		return tracker.getTracked();
 	}
 
 	@Override
 	public boolean isEmpty() {
-		return getTracker().isEmpty();
+		return tracker.isEmpty();
 	}
 
 	@Override
 	public int size() {
-		return getTracker().size();
+		return tracker.size();
 	}
 
 	@Override
 	public S waitForService(long timeout) throws InterruptedException {
-		return getTracker().waitForService(timeout);
+		return tracker.waitForService(timeout);
 	}
 
-	private ServiceTracker<S, S> getTracker() {
-		return trackServices.tracker();
+	private class InnerCustomizer implements ServiceTrackerCustomizer<S, S> {
```

Comment:
Should this be a static inner class? I don't think it depends upon anything in the outer class.

Reply:
fixed